### PR TITLE
New duty_time sensor

### PR DESCRIPTION
**Pull request by dudanov**
## Description:

New `Duty Time` component.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5069

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/duty_time.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+
+Configuration variables:
+------------------------
+
+- **sensor** (*Optional*, :ref:`config-id`): The ID of the ``binary_sensor`` to track the duty time. *May not be
+  used with* ``lambda``.
+- **lambda** (*Optional*, :ref:`lambda <config-lambda>`): Lambda that will be called in a loop to get the current
+  state of the tracked object. *May not be used with* ``sensor``.
+- **last_time** (*Optional*): Information of the last switch-on time sensor.
```
Comment (jesserockz) [inline components/sensor/duty_time.rst:40]:
```suggestion
- **last_time** (*Optional*): Information of the last switch-on time sensor.
  All options from :ref:`Sensor <config-sensor>`.
```

File: index.rst [hunk 2]
```diff
@@ -160,6 +160,7 @@ Core
     Integration, components/sensor/integration, sigma.svg
     MQTT Subscribe, components/sensor/mqtt_subscribe, mqtt.png
     Uptime Sensor, components/sensor/uptime, timer.svg
+    Duty Time, components/sensor/duty_time, timer-play-outline.svg
```
Comment (jesserockz) [inline index.rst]: This probably should go under `miscellaneous`

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: index.rst (hunk 2 above)
Comment (dudanov) [reply index.rst:163]: Added to this category by analogy with the `Uptime` component. I don't see a fundamental difference. Why is `Uptime` in the `Core` category?

Comment (dudanov) [reply index.rst:163]: Jesse, here I will go on principle, or we will move all the controversial components `Uptime`, `Integration`, `WiFi Signal` to the category `miscellaneous` or leave it as it is. I'm sorry, but I don't see the difference between them. Why are they in this category? Should there be some kind of logic? :)

Comment (dudanov) [reply index.rst]: In general, at first I was indignant that `Uptime` was also in the `core` category, by analogy with which I added my component here. But then I decided to take the `Integration` component into the `miscellations` too.

**Review by jesserockz — APPROVED** [review]
Thanks @dudanov

**Pull request merged**